### ISSUE TITLE
DNS66 not blocking any ads.

### ISSUE THREAD

**Issue by RandomUser202** [julian-klode/dns66#89]
I recently installed the app from F-Droid and tried to use it. But once I refreshed the host files, restarted the app and ran it, it didn't work. I tried out a few sites I'll use to test ad blocking such as 4Chan and Forbes and ads still popped up. I disabled any data restricting settings I had on and nothing. Ads still seem to just go through no matter what.

**Comment by jerryn70** [comment]
Post the complete link, i will check

**Comment by arynadam** [comment]
If you are using Chrome, disable browser compression (data saver)..

**Comment by anoadragon453** [comment]
![screenshot_20170306-002642](https://cloud.githubusercontent.com/assets/1342360/23602129/b349fbd2-0203-11e7-80a5-1b4990b786cf.png)


I'm using Chromium with DNS66 enabled full-time without data saver and am still getting ads. Here is one ad I found on [this page](https://m.phys.org/news/2017-03-nasa-magnetic-shield-mars-atmosphere.html):

https://googleads.g.doubleclick.net/aclk?sa=l&ai=CG95Idxu9WKvBB4zb-QOUhY7ICI_q__1E-8ntoNgEhqe6poUBEAEg9raFAmDJruuGyKOMG6AB-J793APIAQKoAwHIA8kEqgSlAU_QlRbGL84F1T2h3oY04VwLcVG6wXgu1FalUrblYI8q2jF5clbGurNt86dNH4tSeDouZDLjj3-n9P8Jsz_SLLGsvhMumkqtgaSvi5lb95LOkQ4MqZubrs4xR3xrLjQ1o8l4DUVZZe0zzaEPuKx993hepAIQpDR1SgtArgM1RoY3WMx02B0lhk8FJfCO-PCa7AYtjMUKWP1jcUUyFV-qBk1X0Aa9CaAGAtgGAoAH8OCCI6gHpr4b2AcB0ggFCIBhEAE&num=1&sig=AOD64_3sv3l4X_jQXhgZxQjAB-Qa8C8FVg&client=ca-pub-0536483524803400&nm=1&nx=72&ny=193&mb=2&bg=!DQ6lDhZE8Y6YsCsRqdMCAAABVVIAAAAkmQFtDIhGQ2R2SxqCc4wCZP9kXvVvOSX4loVl44-9OrklK2TPfinewnPxEnPOuDIpiA-OLRoO7eRiVSMygtgJuntUC29e9pGp5BvSgh1RcXCXypDldRFWl5kg0qJEK0v1b1w32OxCfzwQM9chudJiNIOUO1xeRJpgjGPDTLNRuBnpaKdrWff7pM1jGjzUNFvGz6TDEDeTrIHLg4zwJxSPqDdOTkdzTGPmQrt5LoLkL4CxS9InXXFxIg3KA31HgbOjeYIpy33DUt3svoHi3Brr-hssPO4A0U72BaCgNC0WIrvxOFo9RbXJUm_jSI6mRxJ9fFZNCtp3BjpXIWCtBi-wxZ9P6vxjSGM_wuwLrkP3cQSUbZIkE-AzoNYb2ghELRe5X7vM_7tmgIRR7r_8zE7_zjbecjzUtCIneVg4tBU-hgcNYCqszQo9CuzcIMpe0vanVbmInGHtTCvOxMcppQ0Ot44I-a5beHgngCPKRbxxgdQ&adurl=http://www.panasas.com/sites/default/files/uploads/docs/storage-manufacturing-buyer-guide.pdf%3Fsrc%3Dweb-cpc

**Comment by aggsol** [comment]
I thought lists are active then they are green?!

**Comment by SMed79** [comment]
You must enable domain filtering manually update the hosts files (using the refresh button).

![dns66 domain filtering](http://i.imgur.com/Pha9Q0n.png)

![dns66 refresh button](http://i.imgur.com/PGSc0Fo.png)

**Comment by aggsol** [comment]
The color coding is not helpfull. Maybe a Pause and Play icon would be clearer.

**Comment by julian-klode** [comment]
Guys. The colour is just a hint. You can just click the item and use the spinner if you are unsure.

And the right one to block something is red. Have you never seen a traffic light?

**Comment by anoadragon453** [comment]
@julian-klode Thanks for the clarification.

If I have all my lists marked as Deny, how come I still see frequent ads? Do the default lists not block google ads?

**Comment by julian-klode** [comment]
Nobody can really say. They should be blocked. If the lists are downloaded and marked red, they really should be.

So, maybe Chromium is circumventing DNS66 in some way. The only way to know this is to look at logcat.

Or well, maybe the Google ads domain lookup was cached somewhere from before you started DNS66, and the cache has not expired yet.

**Comment by anoadragon453** [comment]
Thanks for the insight. I'll try to get a logcat in the next few days to see what might be the root cause.

My lists should be up to date and I would be surprised if it was a cache issue since I've been running DNS66 for about 3 days straight now. 

Hopefully the logs will provide some answers.

**Comment by smichel17** [comment]
I am not sure if it is the same issue, but for me, DNS66 stops blocking ads after running for a long time.

- I don't know how long you have to run DNS66 for the blocking to stop working. It is less than 24 hours.
- I have never been able to get blocking to stop working while actively using my phone, although I also cannot verify that inactivity is required for blocking to stop working.
- I can reliably reproduce this, so @julian-klode if you want to make a branch with a lot of debug logs, I can build and install on my device and get you a logcat (or if there is enough debugging enabled on the release version, I can get a logcat from that).

**Comment by julian-klode** [comment]
@smichel17 There should already be more debugging info in the logs than needed :D I guess Android kills it at some point, I've seen it too. But I only notice this a few hours later and reading hours of logcat is painful, even if still existed (there's a limit on the length AFAICT).

**Comment by smichel17** [comment]
> @smichel17 There should already be more debugging info in the logs than needed :D I guess Android kills it at some point, I've seen it too.

It's kind of odd because when this does happen, the notification (both the system notification about VPN and DNS66's notification, if enabled) is still showing. If DNS66 were killed, I'd expect its notification to be dismissed, or at the least, dismissable.

>But I only notice this a few hours later and reading hours of logcat is painful, even if still existed (there's a limit on the length AFAICT).

You could get around this by logging to a db.

**Comment by julian-klode** [comment]
@smichel17 Ah, if the notification stays on, it's very likely an issue in DNS66. Actually just re-discovered that the whole stops working issue is already tracked in #36. If you can grab a logcat as soon as you notice that this happened, that might be vastly useful.

That said, the reconnecting code might indeed be a bit weird. I assume it's somewhere in there- In the hostdb branch I'm rewriting this (gotta split this out of there, it's unrelated to host db itself; it just does not reconnect there anymore, apparently because we're reconnecting to fast for Android).

**Comment by MurkBRA** [comment]
just download the last version
https://github.com/julian-klode/dns66/releases/latest
today is v0.3.1
I download from F-Droid  and got this old version 0.2.2 and simple don't work...

**Comment by anoadragon453** [comment]
I'm on 0.3.0 from F-droid. There doesn't seem to be anything substantial in the point release that would help block ads better. Not sure.

**Comment by RandomUser202** [comment]
I finally remembered my password for this throwaway account. Anyway, I updated the app and now everything seems to be working just fine. Blocks ads like normal now. Some *ahem* Not Safe for Work sites seem to slip by but works great none the less.

**Comment by RandomUser202** [comment]
Hmm, when I used Naked Browser with DNS66, it seemed to block everything perfectly. Ads slipped through in Chromium.

**Comment by MurkBRA** [comment]
Like arynadam commented
> If you are using Chrome, disable browser compression (data saver)..

**Comment by anoadragon453** [comment]
Same thing, getting occasional ads (sometimes they're blocked, sometimes not, VPN on 24/7) in Chromium, data saving disabled.

**Comment by smichel17** [comment]
@anoadragon453 If it is working some of the time, your issue is probably #36, not this.

**Comment by MurkBRA** [comment]
This addon work with some public list (hosts file) with address of ads but is impossible list all address of internet because she is always growing. you can search for new hosts list or create you own and add some unlisted ads address.
I can recomend this [here](https://github.com/StevenBlack/hosts/)
> This repository consolidates several reputable hosts files, and merges them into a unified hosts file with duplicates removed. This repo provides several hosts files tailored to you need to block.

**Comment by anoadragon453** [comment]
Battery optimization is and has been off. I will say that testing a few news sites I do not see any ads at the moment.

In my current location the Wi-Fi is sort of spotty and I am switching back and forth between Wi-Fi and LTE frequently, forcing DNS66 to have to keep reconnecting.

I'll monitor for any ads over the news few days and report back if I see any.

**Comment by anoadragon453** [comment]
Yep just got the following ad:

```
https://adclick.g.doubleclick.net/pcs/click?xai=AKAOjsvvnHh9bBSoq9ae1WEYncTURGSxJ3nqC0sYlapauOfmf9Ac_JrS_6_DNem7HBB2huTYeaqn8BWIygG5ocY5Ap_Bk3DO_6LL2eSRTBs6vaIMPckvYE__26efjKUo&sig=Cg0ArKJSzHhJIkrQ3a-H&urlfix=1&adurl=https://adclick.g.doubleclick.net/pcs/click%3Fxai%3DAKAOjsvD-mC2bmzmvbaVyCwaLnpbtMuId979KrulbGUP5Q_z4M2AUx97FWkaoUxzgn5iZ7209LAMq7EdH2Mq4w4Z9AW79T9kevnw_Lvc-vlwvaB2XJR1YUCNbc-b1yuavxh0mYMfXc3Z4dcuBqxksM8mvPPhHOu4QFgNgIoBUqVEGiPMGnJqWNlX45N5a8m4DHQnGyGzSKZBCAJCL3JXgQCEcDBuhnxcGdRSlYicsSrhUvcvBLhy_6kowomu8yyjnOiFoQ%26sig%3DCg0ArKJSzKM8LEcj1-IqEAE%26urlfix%3D1%26adurl%3Dhttp://www.walmart.com/ip/Listerine-Cool-Mint-Antiseptic-Mouthwash-1.5-L/871623

```

DNS66 was enabled full time. Got in Chromium Incognito mode on this website: http://www.webmd.com/oral-health/wisdom-tooth-extraction

**Comment by RandomUser202** [comment]
Fuck me, tapped on Close issue by accident.

**Comment by julian-klode** [comment]
@aggsol While rereading this bug, I read your comment about the play and pause icon. That sort of makes sense. But while play/pause are a nice analogy, they are also fairly confusing: Does a pause icon indicate that stuff is paused, or that is running? For a media player, it would be the latter, but that would not even work in a three-valued toggle.

My initial idea was to use street signs (STOP is a pretty good sign here, as are do-not-enter-style signs). But the "allow" and "ignore" states don't really work that well.

I think there will always be people confused by the icon choice, but at least 0.4 now has a text at the top that explains it.

**Comment by julian-klode** [comment]
@RandomUser202 @anoadragon453 Please check if it is still a problem with 0.4 when you have time (and it is available on fdroid, if you use f-droid :D). It could be related to some IPv6 changes.

**Comment by anoadragon453** [comment]
@julian-klode Thanks Julian, F-droid still offers 0.3.1 in their official repos, though I'll test 0.4 as soon as it's available!

**Comment by VikasDanny** [comment]
Can we have something like when the device is idle, turn the VPN off to save more charge/power/battery/whatever? Then when you wake it up, the VPN gets automatically turned on. I am sure this is possible. Is it too much for an open source or a free app? 😆

**Comment by smichel17** [comment]
When I use spotify, I often have my device's screen off, so I would not want DNS66 to stop working then.

**Comment by VikasDanny** [comment]
But how would the ads bother you if the screen is off?

**Comment by anoadragon453** [comment]
Presumably this would just be an option, so the user can choose based on their use case.

**Comment by smichel17** [comment]
> But how would the ads bother you if the screen is off?

Audio ads.

**Comment by julian-klode** [comment]
Please stop talking about device-off stuff. That is getting out of control, it is entirely irrelevant to this issue. And it's not going to happen anyway. DNS66 requires basically no power (if there are no DNS requests, it is waiting, it's not even running - it waits for the poll() syscall to return). Turning it off and on again will very likely eat a lot resources and save nothing at all.

**Comment by julian-klode** [comment]
@anoadragon453 is this still an issue for you in 0.4 or 0.4.1?

**Comment by anoadragon453** [comment]
@julian-klode I actually stopped using the app unfortunately, so not sure.

**Comment by SamHH** [comment]
I installed DNS66 on my partner's phone (unrooted Samsung Note 4 on MM) and it didn't seem to have any effect. The service is enabled and I'm 99.9% sure I enabled filtering. Assuming I did, what steps should I take to help debug the issue?

**Comment by jebeld17** [comment]
Similar thing here. It is blocking *some* ads, but not a whole ton... not nearly as many as a paid app like AdGuard.

I think a good way to help implement better ad-blocking is to have a "crowdsourced" mode that enables collection of user-consented blacklists and whitelists to add to future updates for everyone.

**Comment by ponchofiesta** [comment]
Only to keep this up2date. I tried DNS66 yesterday and it blocked nothing for me in Chrome (clean install, data saver disabled). Typically I'm using self compiled Chromium with integrated ad blocking, which blocks ads fine but only in Chromium.

I also tried Virtual Hosts app, which seems to be the same type of app. It also didn't block anything.

**Comment by muvvag** [comment]
I am using an app - Indian constitution by philoid developers and the dns66 is not blocking ads from the app.

**Comment by smichel17** [comment]
@muvvag if *some* ads are blocked, this is a different issue and you should open a new issue. I'll also preemptively mention, DNS66 blocks based on user-configurable lists. It ships with some default lists, but @julian-klode is not involved in maintaining those lists. That is to say, if the problem is "Some domains are not on the list", it is not a bug with DNS66; the solutions are "add a list with that domain" or "file a bug with the list maintainers so they can add the domain to their list".

@julian-klode I think you should close this issue, as it's way too long and contains too many unrelated issues to be useful (not to mention, the original one, caused by app standby and/or doze, is resolved, I think). If there are other outstanding bugs, people can file new issues.

**Comment by sa7hvik** [comment]
Hi All,

I recently installed dns66 and i am trying to block ads generated by the apps installed not by browsing via chrome or some web browser. Looks like it is not blocking any ads, as ads keeps on popping up every 5 seconds still.

Pls help me in adjusting i could do on Dns66 to block the ads.

Thanks in advance.

Cheers,
Sathvik

**Comment by badmojr** [comment]
@sa7hvik
Try adding this hosts file to DNS66.
http://1hosts.cf

**Comment by jebeld17** [comment]
I gave up on the project and went back to AdGuard. It's a paid service, but it works.

**Comment by smichel17** [comment]
Unsollicited advertising in a bug tracker feels like pretty poor taste to me.

**Comment by pm1111** [comment]
If the problem of ads slipping through isn't caused by DNSSEC it might be caused by DNS over TLS invented by Google and implemented on their name servers. It uses port 853 and is described in RFC 7858

**Comment by Cassolotl** [comment]
I just installed DNS66, enabled filtering, refreshed to update the lists, enabled ad-blocking in Instagram and Twitter, and ads are still getting through. I restarted my phone and that didn't affect anything. Are there particular lists I need to add to block ads in those apps?

**Comment by phoerious** [comment]
For me it stopped working entirely after I upgraded my One Plus to Android 9. I tried reinstalling the app, updated all filter lists, but no ads are blocked at all.

**Comment by phoerious** [comment]
Any news on this? I'm considering un installing dns666 and rooting my phone, since it's entirely useless at the moment. I'm not sure what's happening here, but it appears like neither the default route is changed nor are there any specific routing rules for ad servers. 

```
:/ $ ip route show table all | grep 'table [0-9]'
default via 192.168.188.1 dev wlan0 table 1029 proto static
192.168.188.0/24 dev wlan0 table 1029 proto static scope link
192.0.2.0/24 dev tun0 table 1132 proto static scope link
192.0.2.3 dev tun0 table 1132 proto static scope link
default dev dummy0 table 1003 proto static scope link
255.0.1.0 dev r_rmnet_data0 table 1020 proto static scope link
2001:470:6d:7ae::/64 dev wlan0 table 1029 proto kernel metric 256 expires 7156sec pref medium
2001:470:6d:7ae::/64 via fe80::be05:43ff:fe91:994d dev wlan0 table 1029 proto ra metric 1024 expires 1756sec pref medium
fe80::/64 dev wlan0 table 1029 proto kernel metric 256 pref medium
fe80::/64 dev wlan0 table 1029 proto static metric 1024 pref mediumdefault via fe80::be05:43ff:fe91:994d dev wlan0 table 1029 proto ra metric 1024 expires 1756sec mtu 1480 hoplimit 255 pref medium
2001:db8::/120 dev tun0 table 1132 proto kernel metric 256 pref medium
2001:db8::/120 dev tun0 table 1132 proto static metric 1024 pref medium
fe80::/64 dev tun0 table 1132 proto kernel metric 256 pref medium
fe80::/64 dev dummy0 table 1003 proto kernel metric 256 pref mediumdefault dev dummy0 table 1003 proto static metric 1024 pref medium
fe80::/64 dev rmnet_data0 table 1009 proto kernel metric 256 pref medium
default via fe80::4404:4caf:b52c:d180 dev rmnet_data0 table 1009 proto ra metric 1024 expires 19819sec hoplimit 255 pref medium
fd95:d1e:12:: dev r_rmnet_data0 table 1020 proto static metric 1024 pref medium
fe80::/64 dev r_rmnet_data0 table 1020 proto kernel metric 256 pref medium
:/ $ ip route show table 1132
192.0.2.0/24 dev tun0 proto static scope link
192.0.2.3 dev tun0 proto static scope link
:/ $
```

**Comment by pufferchung** [comment]
I love DNS66 but had to uninstall it and switched to ADguard

**Comment by sodiumjoe** [comment]
There's a thread [here](https://github.com/julian-klode/dns66/issues/289#issuecomment-420080183) for fixing a chrome issue that worked for me.

**Comment by phoerious** [comment]
Is it really just Chrome not respecting DNS settings? I've got ads slipping through in a couple of other apps as well.

**Comment by sodiumjoe** [comment]
@phoerious could be a separate issue, maybe those ads are not blocked by your hosts files. But switching to chrome beta and turning off the async dns setting worked for me.

**Comment by phoerious** [comment]
Not for me. No matter what settings and DNS servers I use, Chrome shows ads.

**Comment by smichel17** [comment]
@julian-klode I think you should close this issue. It's so long that it's near-impossible to get useful info from, and it's likely that there's a lot of different problems reported here.

**Comment by phoerious** [comment]
Alright, with the current beta and disabling async DNS in chrome://flags, it's working now. Stable chrome still not updated, though.

**Comment by patrislav1** [comment]
Disabling async DNS in `chrome://flags` also fixed it for me. BTW, that flag is now also there in regular chrome (70.0.3538.80), at least on my Android9 Pixel

**Comment by phoerious** [comment]
Yes, but it doesn't work somehow. I have disabled Async DNS in both Chrome Stable and Chrome Beta, but adblocking still doesn't work in Stable, although it's version 70 now.

**Comment by demetry14** [comment]
What I have noticed is that the Google Play Store on Android is not blocked and therefore all apps seem to get an ad channel through, but when I actively seeking out Play Store in the block list and deselect it then ads are fully blocked.

But I have to keep DNS66 open in my active windows otherwise Play Store re-enables VPN bypass and ads return.

**Comment by tegutistgut** [comment]
For me, dns66 doesn't blocks ads in games (happy wheels) and apps. But adguard does.

**Comment by alexanderadam** [comment]
@tegutistgut are all lists activated?
Maybe you want to extend your host list with more entries, like the ones from [Adguard](https://github.com/AdguardTeam/AdguardFilters). Also [blokada has more lists](https://github.com/blokadaorg/blokadaorg.github.io/blob/master/api/v4/canonical/defaults/filters.txt) that can be activated.
And if none of those help, you can also add manual rules.
IMHO all of the adblockers are just simply follow the rules that you have activated.

**Comment by tegutistgut** [comment]
@alexanderadam hi thanks for the ideas.
I couldn't understand how I add the adguard filters. And yes, I activated all filters and it didn't work.
Manual rules also don't work. (Https and http). IDK what to do now.

**Comment by alexanderadam** [comment]
> Manual rules also don't work. (Https and http)

So if you go to "Hosts" tap on the "+", select Action "Deny" and add "www.example.org", you can still enter "www.example.org" without problems? :thinking:

**Comment by tegutistgut** [comment]
I can't visit the site with my browser, but the ad banner still pops up in the game...

**Comment by alexanderadam** [comment]
Well that means DNS66 is working perfectly fine.
It just doesn't know that you want to have a particular host blocked. DNS66 can't automatically figure every ad-serving hostname out. This is why you have the possibility to add other hosts you want to have blocked. So you have to find a way to find out which host is giving out your particular ads and add it to the blocking list. Blokada and [personaldnsfilter](https://github.com/IngoZenz/personaldnsfilter) have an overview on accessed hostnames, which might help you in finding out that host.

I can't help you with that though because I'm not playing "happy wheels".
Your problem is offtopic anyway though, because blocking in general is working for you, as you just wrote.

**Comment by tegutistgut** [comment]
Yes, it worked again.
Thank you, for helping me

**Comment by smed79** [comment]
@tegutistgut 

Find here [Ultimate-Hosts-Blacklist](https://github.com/Ultimate-Hosts-Blacklist) some hosts for different purpose 
(add the raw hosts file from the folder "output > hosts > ACTIVE").

A good alternative is to 
- enable custom DNS 
- disable the default then use [adgurad dns](#119) (https://i.ibb.co/ng0bCJ1/123.png)
